tools/compile_tf_graph.py

```diff
@@ -197,7 +294,7 @@ class StateVar:
     Represents a state variable, i.e. either a state, a choice, or encoder state, etc.
     """
 
-    def __init__(self, parent, name, initial_value, data_shape):
+    def __init__(self, parent, name, initial_value, data_shape, choice_dependent):
```

Comment:
I don't understand why `choice_dependent` is needed? It should not be needed. Keep the code simple for now.

Reply:
why this one param makes it so complicated, it saves a lot of RAM when I use a large beam

Reply:
See the other discussion why it becomes complicated. At least if you want to have it correct. But I would definitely be against some incorrect or not generic solution here. This would also become terrible for debugging later, e.g. some new config where some of your assumptions become wrong, but this script still silently returns sth, but the Sprint search behaves buggy then, e.g. also returns sth, but not correctly.